src/CMakeLists.txt

```diff
@@ -1 +1,265 @@
+  YButtonBox.cc
+  YCheckBox.cc
+  YCheckBoxFrame.cc
+  YComboBox.cc
+  YContextMenu.cc
+  YDateField.cc
+  YDialog.cc
+  YDialogHelpers.cc
+  YDownloadProgress.cc
+  YDumbTab.cc
+  YEmpty.cc
+  YFrame.cc
+  YImage.cc
+  YInputField.cc
+  YIntField.cc
+  YItemSelector.cc
+  YLabel.cc
+  YLayoutBox.cc
+  YLogView.cc
+  YMenuBar.cc
+  YMenuButton.cc
+  YMenuWidget.cc
+  YMultiLineEdit.cc
+  YMultiProgressMeter.cc
+  YMultiSelectionBox.cc
+  YPackageSelector.cc
+  YGraph.cc
+  YPartitionSplitter.cc
+  YProgressBar.cc
+  YPushButton.cc
+  YRadioButton.cc
+  YRadioButtonGroup.cc
+  YReplacePoint.cc
+  YRichText.cc
+  YSelectionBox.cc
+  YSlider.cc
+  YSpacing.cc
+  YSquash.cc
+  YTable.cc
+  YTimeField.cc
+  YTimezoneSelector.cc
+  YTree.cc
+  YWidget.cc
+  YWizard.cc
+
+  FSize.cc
+  YStringTree.cc
+  YRpmGroupsTree.cc
+
+  YPopupInternal.cc
+  YPropertyEditor.cc
+  )
+
+
+set( HEADERS
+  YUI.h
+  YApplication.h
+  YWidgetFactory.h
+  YOptionalWidgetFactory.h
+  YSettings.h
+  YPath.h
+
+  YBuiltinCaller.h
+  YBothDim.h
+  YChildrenManager.h
+  YColor.h
+  YCommandLine.h
+  YDescribedItem.h
+  YDialogSpy.h
+  YEvent.h
+  YEventFilter.h
+  YEnvVar.h
+  YItem.h
+  YItemCustomStatus.h
+  YIconLoader.h
+  YMacro.h
+  YMacroPlayer.h
+  YMacroRecorder.h
+  YMenuItem.h
+  YPackageSelectorPlugin.h
+  YGraphPlugin.h
+  YProperty.h
+  YShortcut.h
+  YShortcutManager.h
+  YSimpleEventHandler.h
+  YSingleChildContainerWidget.h
+  YTableHeader.h
+  YTableItem.h
+  YTreeItem.h
+  YTypes.h
+  YUIException.h
+  YUILoader.h
+  YUILog.h
+  YUIPlugin.h
+  YUISymbols.h
+  YWidgetID.h
+  YWidget_OptimizeChanges.h
+
+  YSelectionWidget.h
+  YSimpleInputField.h
+
+  YAlignment.h
+  YBarGraph.h
+  YBusyIndicator.h
+  YButtonBox.h
+  YCheckBoxFrame.h
+  YCheckBox.h
+  YComboBox.h
+  YContextMenu.h
+  YDateField.h
+  YDialog.h
+  YDownloadProgress.h
+  YDumbTab.h
+  YEmpty.h
+  YFrame.h
+  YImage.h
+  YInputField.h
+  YIntField.h
+  YItemSelector.h
+  YLabel.h
+  YLayoutBox.h
+  YLogView.h
+  YMenuBar.h
+  YMenuButton.h
+  YMenuWidget.h
+  YMultiLineEdit.h
+  YMultiProgressMeter.h
+  YMultiSelectionBox.h
+  YPackageSelector.h
+  YGraph.h
+  YPartitionSplitter.h
+  YProgressBar.h
+  YPushButton.h
+  YRadioButtonGroup.h
+  YRadioButton.h
+  YReplacePoint.h
+  YRichText.h
+  YSelectionBox.h
+  YSlider.h
+  YSpacing.h
+  YSquash.h
+  YTable.h
+  YTimeField.h
+  YTimezoneSelector.h
+  YTree.h
+  YWidget.h
+  YWizard.h
+
+  FSize.h
+  ImplPtr.h
+  TreeItem.h
+  YRpmGroupsTree.h
+  YStringTree.h
+  YTransText.h
+  YExternalWidgetFactory.h
+  YExternalWidgets.h
+  )
+
+
+
+# Add shared lib to be built
+add_library( ${TARGETLIB} SHARED ${SOURCES} ${HEADERS} )
+
+# Libraries that are needed to build this shared lib
+#
+# Not sure if we really need to link against Boost::boost;
+# the classes used here may all be in header files only without
+# any binary part that needs linking. But better safe than sorry.
+target_link_libraries( ${TARGETLIB} dl pthread Boost::boost )
```

Comment:
Huh, what does the `::` syntax mean? It's hard to search for so a comment would be useful.

Reply:
I admit I stole that from another place where it was used. AFAIK this links only against one of many available Boost libraries; that's what the CMake ref doc suggests to use:

  https://cmake.org/cmake/help/latest/module/FindBoost.html

https://cliutils.gitlab.io/modern-cmake/chapters/packages/Boost.html

> The import targets are in the Boost:: namespace. Boost::boost is the header only part. The other compiled libraries are available, and include dependencies as needed.

This seems to imply that we don't need to add `Boost::boost` to the libs that we link against.

Reply:
It turns out we don't need to link against `Boost::boost`, so I removed it, and it still works.
Changed the comment accordingly in case we ever need to add a Boost lib.